engine/src/main/java/org/terasology/engine/network/internal/ServerConnectListManager.java

```diff
@@ -28,14 +28,24 @@ public class ServerConnectListManager {
     private static final Gson GSON = new Gson();
 
     private Context context;
-    private Set<String> blacklistedIDs;
-    private Set<String> whitelistedIDs;
-    private final Path blacklistPath;
-    private final Path whitelistPath;
+    private Set<String> denylistedIDs;
+    private Set<String> allowlistedIDs;
+    private final Path denylistPath;
+    private final Path allowlistPath;
 
     public ServerConnectListManager(Context context) {
-        blacklistPath = PathManager.getInstance().getHomePath().resolve("blacklist.json");
-        whitelistPath = PathManager.getInstance().getHomePath().resolve("whitelist.json");
+        // although this seems redundant, compiler wouldn't accept assigning then checking
+        if (Files.exists(PathManager.getInstance().getHomePath().resolve("denylist.json"))) {
+            denylistPath = PathManager.getInstance().getHomePath().resolve("denylist.json");
+        } else {
+            denylistPath = PathManager.getInstance().getHomePath().resolve("blacklist.json");
+        }
+        // although this seems redundant, compiler wouldn't accept assigning then checking
+        if (Files.exists(PathManager.getInstance().getHomePath().resolve("allowlist.json"))) {
+            allowlistPath = PathManager.getInstance().getHomePath().resolve("allowlist.json");
+        } else {
+            allowlistPath = PathManager.getInstance().getHomePath().resolve("whitelist.json");
+        }
```

Comment:
```suggestion
        // Check for old naming first to retain compatiblity, otherwise use new naming by default.
        Path potentialDenylistPath = PathManager.getInstance().getHomePath().resolve("blacklist.json");
        Path potentialAllowlistPath = PathManager.getInstance().getHomePath().resolve("whitelist.json");
        if (!Files.exists(potentialDenylistPath)) {
            potentialDenylistPath = PathManager.getInstance().getHomePath().resolve("denylist.json");
        }
        if (!Files.exists(potentialAllowlistPath)) {
            potentialAllowlistPath = PathManager.getInstance().getHomePath().resolve("allowlist.json");
        }
        denylistPath = potentialDenylistPath;
        allowlistPath = potentialAllowlistPath;
```

`denylistPath` and `allowlistPath` are final variables, so they can only be assigned once in the constructor. Using a temporary variable here should work (but I wrote this quickly so please test that it compiles and works).